src/pilot/assign_role.py

```diff
@@ -901,26 +846,16 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index, os_volume_size_gb,
     if role_index.index:
         role = "node:{}-{}".format(flavor, role_index.index)
 
-    if 'capabilities' in node.properties:
-        value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
-        LOG.info(str(node.properties))
-        LOG.info(str(value))
-    else:
-        value = "{},boot_option:local".format(role)
-        LOG.info("..!")
+    value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
+    LOG.info(str(node.properties))
+    LOG.info(str(value))
```

Comment:
As a follow-up to this pull request, let's make this work when out-of-band (OOB) inspection or in-band (IB) introspection has not been performed before `assign_role.py` is run. That workflow is used by the Ironic Team when integration testing their changes. In that case, the `capabilities` property may not be present and line 849 above would cause an exception to be raised.

The new implementation could be something like the following:

    if 'capabilities' in node.properties:
        value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
    else:
         value = "{},boot_mode:uefi,boot_option:local".format(role)

    LOG.info(str(node.properties))
    LOG.info(str(value))